guides/micronaut-data-jdbc-repository/java/src/main/java/example/micronaut/GenreRepository.java

```diff
@@ -0,0 +1,67 @@
+package example.micronaut;
+
+import example.micronaut.domain.Genre;
+import io.micronaut.data.annotation.Id;
+import io.micronaut.data.exceptions.DataAccessException;
+import io.micronaut.data.jdbc.annotation.JdbcRepository;
+import io.micronaut.data.model.query.builder.sql.Dialect;
+import io.micronaut.data.repository.CrudRepository;
+
+import javax.transaction.Transactional;
+import javax.validation.constraints.NotBlank;
+import javax.validation.constraints.NotNull;
+import java.util.Collections;
+import java.util.List;
+import java.util.Optional;
+
+@JdbcRepository(dialect = Dialect.H2) // <1>
+public interface GenreRepository extends CrudRepository<Genre, Long> { // <2>
+
+    Optional<Genre> findById(@NotNull Long id);
+
+    Genre save(@NotBlank String name);
+
+    @Transactional
+    default Genre saveWithException(@NotBlank String name) {
+        save(name);
+        throw new DataAccessException("test exception");
+    }
+
+    void deleteById(@NotNull Long id);
+
+    List<Genre> findAll();
+
+    List<Genre> findAllOrderById();
+
+    List<Genre> findAllOrderByIdDesc();
+
+    List<Genre> findAllOrderByName();
+
+    List<Genre> findAllOrderByNameDesc();
+
+    default List<Genre> findAll(@NotNull SortingAndOrderArguments args) {
```

Comment:
Can this be rewritten to use `Pageable`?

Reply:
Should in fact extend from `PageableRepository` and then you get the method for free

Reply:
I've extended from `PageableRepository` and improved the repository section copy.